test/Libraries/Microsoft.AspNetCore.HeaderParsing.Tests/HeaderParsingFeatureTests.cs

```diff
@@ -147,10 +147,11 @@ public void TryParse_returns_false_on_error()
 
         Assert.Equal("Can't parse header 'Date' due to 'Unable to parse date time offset value.'.", _logger.Collector.LatestRecord.Message);
 
-        var latest = metricCollector.LastMeasurement!;
+        var latest = metricCollector.LastMeasurement;
+        Assert.NotNull(latest);
         latest.Value.Should().Be(1);
-        latest.Tags["HeaderName"].Should().Be("Date");
-        latest.Tags["Kind"].Should().Be("Unable to parse date time offset value.");
+        latest.Tags["header.name"].Should().Be("Date");
+        latest.Tags["error.type"].Should().Be("Unable to parse date time offset value.");
```

Comment:
We use a closed list of errors (or a full exception name) is other places where `error.type` is set and should follow the same pattern consistently.
The exception message has high-cardinality and should not be used on metrics dimensions.

Reply:
Header parsers should provide a closed set of parsing errors. @geeknoid can you confirm that?